compiler/llvmutil/wordpack.go

```diff
@@ -42,6 +42,24 @@ func EmitPointerPack(builder llvm.Builder, mod llvm.Module, config *compileopts.
 		// it in an alloca first for bitcasting (store+bitcast+load).
 		packedAlloc, _, _ = CreateTemporaryAlloca(builder, mod, packedType, "")
 	} else {
+		// Check if the values are all constants.
+		constant := true
+		for _, v := range values {
+			if !v.IsConstant() {
+				constant = false
+				break
+			}
+		}
+
+		if constant {
+			// The data is known at compile time, so store it in a constant global.
+			global := llvm.AddGlobal(mod, packedType, "")
+			global.SetInitializer(ctx.ConstStruct(values, false))
+			global.SetGlobalConstant(true)
```

Comment:
None of the uses of `EmitPointerPack` rely on this pointer being unique (I checked). Therefore you can mark this `unnamed_addr` using `SetUnnamedAddr`. This will let the compiler merge identical globals.